android/app/src/main/java/com/mattermost/rnbeta/CustomPushNotification.java

```diff
@@ -105,22 +108,23 @@ public void onReceived() throws InvalidNotificationException {
 
         if (channelId != null) {
             notificationId = channelId.hashCode();
-            Object objCount = channelIdToNotificationCount.get(channelId);
-            Integer count = 1;
-            if (objCount != null) {
-                count = (Integer)objCount + 1;
-            }
-            channelIdToNotificationCount.put(channelId, count);
 
-            Object bundleArray = channelIdToNotification.get(channelId);
-            List list = null;
-            if (bundleArray == null) {
-                list = Collections.synchronizedList(new ArrayList(0));
-            } else {
-                list = Collections.synchronizedList((List)bundleArray);
+            synchronized (channelIdToNotificationCount) {
+                Integer count = channelIdToNotificationCount.get(channelId);
+                if (count == null) {
+                    count = 0;
+                }
+
+                channelIdToNotificationCount.put(channelId, count + 1);
```

Comment:
Just for ease of reading it, so people doesn't get confused when value is 1 instead of 0 due to reading too quick.
```suggestion
                     count = 1;
                } else {
                     channelIdToNotificationCount.put(channelId, count + 1);
               }
```

Reply:
I'll move the `count + 1` onto it's own line, but we want to put that value into the map either way, so I'll do it slightly differently than what you've suggested